### DISCUSSION TITLE
Characteristic vs Switch

### DISCUSSION THREAD

**Discussion by BasileD** [SeydX/homebridge-fritz-platform#271] [Q&A]
Hey,

First of all thanks a lot for this awesome homebridge plugin. 

However, I do not seem to understand the functionality of "Characteristic" (instead of "switch"). How can that be used for HomeKit? Unfortunately, I did not find an answer to that yet.

Once again, thank you very much.

All the best.

**Comment by knikkabokka** [comment]
I'm not entirely sure, but my understanding of the difference is this: 

I'll use "DECT" as an example below:

- If you set up the function as a *Switch*, then you can switch this function via Homebridge. (e.g. "Switch DECT on / Switch DECT off)
- If you set up the function as a *Characteristic*, then you can only read the status of the function to use it e.g. in an automation. (e.g. "If DECT is switched off at the Fritz!Box, then set the light in the hallway to red"), but you cannot switch DECT on or off via Homebridge but only directly at the Fritz!Box.
